server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -662,21 +678,33 @@ public Field getRangeField(String name, Range r) {
             }
             @Override
             public Query withinQuery(String field, Object from, Object to, boolean includeFrom, boolean includeTo) {
-                return FloatRange.newWithinQuery(field,
-                    new float[] {includeFrom ? (Float)from : Math.nextUp((Float)from)},
-                    new float[] {includeTo ? (Float)to : Math.nextDown((Float)to)});
+                return createQuery(field, from, to, includeFrom, includeTo,
+                        (f, t) -> FloatRange.newWithinQuery(field, new float[] { f }, new float[] { t }));
             }
             @Override
             public Query containsQuery(String field, Object from, Object to, boolean includeFrom, boolean includeTo) {
-                return FloatRange.newContainsQuery(field,
-                    new float[] {includeFrom ? (Float)from : Math.nextUp((Float)from)},
-                    new float[] {includeTo ? (Float)to : Math.nextDown((Float)to)});
+                return createQuery(field, from, to, includeFrom, includeTo,
+                        (f, t) -> FloatRange.newContainsQuery(field, new float[] { f }, new float[] { t }));
             }
             @Override
             public Query intersectsQuery(String field, Object from, Object to, boolean includeFrom, boolean includeTo) {
-                return FloatRange.newIntersectsQuery(field,
-                    new float[] {includeFrom ? (Float)from : Math.nextUp((Float)from)},
-                    new float[] {includeTo ? (Float)to : Math.nextDown((Float)to)});
+                return createQuery(field, from, to, includeFrom, includeTo,
+                        (f, t) -> FloatRange.newIntersectsQuery(field, new float[] { f }, new float[] { t }));
+            }
+            
+            private Query createQuery(String field, Object from, Object to, boolean includeFrom, boolean includeTo,
+                    BiFunction<Float, Float, Query> querySupplier) {
+                if ((Float) from > (Float) to) {
+                    // wrong argument order, this is an error the user should fix
+                    throw new IllegalArgumentException("Range query `from` value (" + from + ") is greater than `to` value (" + to + ")");
+                }
+                Float correctedFrom = (Float) from + (includeFrom ? 0 : 1);
```

Comment:
See below comment for doubles ranges; floats should also use `nextUp()` instead of just increasing by 1.

Reply:
I think I'll try creating only one  createQuery function...